### PR TITLE
generate date range project report

### PR DESCRIPTION
**Pull request by dchhabda**
*Issue number of the reported bug or feature request: #<number>*

**Describe your changes**
A clear and concise description of the changes you have made.

**Testing performed**
Describe the testing you have performed to ensure that the bug has been addressed, or that the new feature works as planned.

**Additional context**
Add any other context about your contribution here.

### CODE REVIEW

File: pybossa/view/projects.py [hunk 1]
```diff
@@ -2870,70 +2871,53 @@ def del_coowner(short_name, user_name=None):
...
-
-@blueprint.route('/<short_name>/projectreport/export')
+@blueprint.route('/<short_name>/projectreport/export', methods=['GET', 'POST'])
 @login_required
+@admin_or_subadmin_required
 def export_project_report(short_name):
-    """Export individual project information in the given format"""
-    project, owner, ps = allow_deny_project_info(short_name)
-    project_report_csv_exporter = ProjectReportCsvExporter()
+    """Assign users to project based on projects data access levels."""
```
Comment (fiorda) [inline pybossa/view/projects.py]: Comment seems wrong

File: test/test_project_report.py [hunk 2]
```diff
@@ -96,28 +64,64 @@ def test_project_report_with_task_details(self):
...
+        """Test project report is generated with deleting existing report zip"""
         self.register()
         self.signin()
         user = user_repo.get(1)
         project = ProjectFactory.create(owner=user)
-        prce = ProjectReportCsvExporter()
-        with patch.object(prce, '_get_csv', _new_get_csv):
-            with assert_raises(Exception):
-                prce._make_zip(project, None)
+        url = '/project/%s/projectreport/export?type=project&format=csv' % project.short_name
```
Comment (fiorda) [inline test/test_project_report.py]: do you need the query parameters?

File: test/test_project_report.py [hunk 3]
```diff
@@ -96,28 +64,64 @@ def test_project_report_with_task_details(self):
...
+        res = self.app.post(url)
```
Comment (fiorda) [inline test/test_project_report.py]: can we add some assertion on the response?

File: pybossa/forms/validator.py [hunk 4]
```diff
@@ -209,3 +209,17 @@ def __call__(self, form, field):
...
+    def __init__(self, message=None):
+        if not message:
+            message = lazy_gettext("Start date must be earlier than end date.")
+        self.message = message
+
+    def __call__(self, form, field):
+        if not (form.start_date.data and form.end_date.data):
+            return
+
+        if (form.start_date.data > form.end_date.data):
```
Comment (fiorda) [inline pybossa/forms/validator.py]: Unnecessary `()`

File: pybossa/view/projects.py [hunk 5]
```diff
@@ -2870,70 +2871,53 @@ def del_coowner(short_name, user_name=None):
...
+    if end_date:
+        kwargs["end_date"] = end_date.strftime("%Y-%m-%dT23:59:59")
 
-    return {'csv': respond_csv}[fmt](ty)
+    try:
+        project_report_csv_exporter = ProjectReportCsvExporter()
+        res = project_report_csv_exporter.response_zip(project, "project", **kwargs)
+        flash("Generate project report request submitted.", 'success')
+        res.status_code = 200
+        res.location = url_for('.settings', short_name=short_name)
```
Comment (fiorda) [inline pybossa/view/projects.py]: what does this do?

Comment (dchhabda) [reply pybossa/view/projects.py]: not reqd. removed

**Pull request merged**